conans/client/build/compiler_id.py

```diff
@@ -0,0 +1,211 @@
+# -*- coding: utf-8 -*-
+
+import os
+import tempfile
+from six import StringIO
+
+from conans.client.runner import ConanRunner
+
+
+GCC = "gcc"
+LLVM_GCC = "llvm-gcc"  # GCC frontend with LLVM backend
+CLANG = "clang"
+APPLE_CLANG = "apple-clang"
+SUNCC = "suncc"
+MSVC = "Visual Studio"
+INTEL = "intel"
+QCC = "qcc"
+
+
+class CompilerId:
+    """
+    compiler identification description, holds compiler name, full version and
+    other important properties
+    """
+    def __init__(self, name, major, minor, patch):
+        self._name = name
+        self._major = major
+        self._minor = minor
+        self._patch = patch
+
+    @property
+    def name(self):
+        return self._name
+
+    @property
+    def major(self):
+        return self._major
+
+    @property
+    def minor(self):
+        return self._minor
+
+    @property
+    def patch(self):
+        return self._patch
+
+    @property
+    def version(self):
+        return "%s.%s.%s" % (self._major, self._minor, self._patch)
+
+    @property
+    def major_minor(self):
+        return "%s.%s" % (self._major, self._minor)
+
+    def __str__(self):
+        return "%s %s" % (self._name, self.version)
+
+    def __repr__(self):
+        return self.__str__()
+
+    def __eq__(self, other):
+        return self.name == other.name and \
+               self.major == other.major and \
+               self.minor == other.minor and \
+               self.patch == other.patch
+
+    def __ne__(self, other):
+        return not self.__eq__(other)
+
+
+UNKWNON_COMPILER = CompilerId(None, None, None, None)
```

Comment:
There is a misspelling, please check this

```suggestion
UNKNOWN_COMPILER = CompilerId(None, None, None, None)
```

Reply:
renamed